kolibri/core/assets/src/views/k-textbox/index.vue

```diff
@@ -0,0 +1,138 @@
+<template>
+
+  <ui-textbox
+    ref="textbox"
+    v-model="currentText"
+    :label="label"
+    :disabled="disabled"
+    :required="required"
+    :invalid="invalid"
+    :error="invalidText"
+    :autofocus="autofocus"
+    :maxlength="maxlength"
+    :autocomplete="autocomplete"
+    :type="type"
+    :enforceMaxlength="true"
+    :floatingLabel="true"
+    @input="updateText"
+    @keydown="emitKeydown"
+    @focus="$emit('focus')"
+    @blur="$emit('blur')"
+  />
+
+</template>
+
+
+<script>
+
+  import uiTextbox from 'keen-ui/src/UiTextbox';
+
+  /**
+   * Handles user input.
+   */
+  export default {
+    name: 'k-textbox',
+    components: { uiTextbox },
+    props: {
+      /**
+       * v-model
+       */
+      value: {
+        type: [String, Number],
+      },
+      /**
+       * Label
+       */
+      label: {
+        type: String,
+        required: true,
+      },
+      /**
+       * Whether or not disabled
+       */
+      disabled: {
+        type: Boolean,
+        default: false,
+      },
+      /**
+       * Whether or not required
+       */
+      required: {
+        type: Boolean,
+        default: false,
+      },
```

Comment:
actually, I think we decided not to use the html `required` attribute, and to always use JS validation instead